pkg/gadgets/internal/socketenricher/bpf/sockets-map.h

```diff
@@ -0,0 +1,52 @@
+/* SPDX-License-Identifier: (GPL-2.0 WITH Linux-syscall-note) OR Apache-2.0 */
+
+#ifndef SOCKETS_MAP_H
+#define SOCKETS_MAP_H
+
+#ifndef SOCKETS_MAP_IMPLEMENTATION
+
+#include <linux/if_ether.h>
+#include <linux/ip.h>
+#include <linux/tcp.h>
+#include <linux/udp.h>
+
+#endif
+
+#ifndef PACKET_HOST
+#define PACKET_HOST		0
+#endif
+
+#ifdef PACKET_OUTGOING
+#define PACKET_OUTGOING		4
+#endif
+
+
+typedef __u32 ipv4_addr;
+
+struct sockets_key {
+	__u32 netns;
```

Comment:
I initially used `__u64` but it does not work:
```
BPF_CORE_READ_INTO(&socket_key.netns, sock, __sk_common.skc_net.net, ns.inum);
```

`ns.inum` is an unsigned int (see [ns_common.h#L12](https://github.com/torvalds/linux/blob/f9ff5644bcc04221bae56f922122f2b7f5d24d62/include/linux/ns_common.h#L12) aka __u32). If we use `__u64`,  `BPF_CORE_READ_INTO` will copy `sizeof(*(dst))` bytes (8 bytes) so it will copy the `refcount_t count` fr the `struct ns_common` too.

Reply:
Perhaps we should switch to u32 for mntns too? It seems a bit weird to use different types for the same data in different places of the code. For sure, to be done in another PR.

Reply:
https://github.com/inspektor-gadget/inspektor-gadget/issues/1215